config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +37,137 @@
+        Map<Dependency, List<Dependency>> dependencyListMap = bundle.getDependencyMap();
+        if (dependencyListMap != null) {
+            int pathIndex = policyNameWithPath.lastIndexOf("/");
+            final String policyName = pathIndex > -1 ? policyNameWithPath.substring(pathIndex + 1) : policyNameWithPath;
+            Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
+            for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
+                final Dependency dependencyParent = entry.getKey();
+                if (dependencyParent.getName().equals(policyName)) {
+                    //Add the policy dependant folders
+                    final Map<String, Policy> entityMap = bundle.getPolicies();
+                    if (!entityMap.isEmpty()) {
+                        final GatewayEntity policyEntity = entityMap.get(policyNameWithPath);
+                        populateDependentFolders(filteredEntityIds, policyEntity);
+                        filteredEntityIds.add(policyEntity.getId());
+                    }
+
+                    //Add the policy dependencies
+                    for (Dependency dependency : entry.getValue()) {
+                        for (Entity entity : entities) {
+                            int index = entity.getName().lastIndexOf("/");
+                            final String entityName = index > -1 ? entity.getName().substring(index + 1) : entity.getName();
+                            if (dependency.getName().equals(entityName) && dependency.getType().equals(entity.getType())) {
+                                filteredEntityIds.add(entity.getId());
+                            }
+                        }
+                    }
+
+                    final List<Entity> filteredEntities = entities.stream().filter(entity -> filteredEntityIds.contains(entity.getId())).collect(Collectors.toList());
+                    entityDependenciesList.addAll(filteredEntities);
+
+                    return entityDependenciesList;
+                }
+            }
+        }
+
+        return entityDependenciesList;
+    }
+
+    private void populateDependentFolders(Set<String> filteredEntityIds, GatewayEntity policyEntity) {
+        if (policyEntity instanceof Folderable) {
+            Folder folder = ((Folderable) policyEntity).getParentFolder();
+            while (folder != null) {
+                filteredEntityIds.add(folder.getId());
+                folder = folder.getParentFolder();
+            }
+        }
+    }
+
+    /**
+     * Creates AnnotatedEntity object by scanning all the annotations and gathering all the information required to
+     * generate the bundle and its metadata.
+     *
+     * @param encass Encapsulated assertion
+     * @param bundleVersion Bundle version
+     * @return AnnotatedEntity
+     */
+    private AnnotatedEntity createAnnotatedEntity(final Encass encass, final String bundleVersion, final String entityName) {
```

Comment:
Looks like this method is made specific to ENCASS. We could have made this method extensible to future changes.
Still, I'm OK.